int32 x
int32 y
int32 width
int32 height
int32 objectType
int32 insideWarningZone
int32 distance